test/lib/middleware.test.js

```diff
@@ -0,0 +1,69 @@
+const chai = require('chai')
+const proxyquire = require('proxyquire')
+const spies = require('chai-spies')
+
+chai.use(spies)
+const expect = chai.expect
+
+const CORRELATION_ID = 'correlation'
+const AUTH_TOKEN = 'authentication'
+
+const mockDomain = {
+  add: () => { },
+  run: callback => { mockDomain.runCallback = callback }
+}
+const session = { correlateSession: () => { } }
+const domain = { create: () => mockDomain }
+const cookie = { parse: () => ({ 'auth-token': AUTH_TOKEN }) }
+let next = () => { }
+const req = {
+  headers: {
+    'correlation-id': CORRELATION_ID,
+    'auth-token': AUTH_TOKEN,
+    cookie: ''
+  }
+}
+const res = { }
+
+session.correlateSession = chai.spy(session.correlateSession)

```

Comment:
You can't define spy in a global level, because the data of the spies would be saved from test to test. See what I did in session.test.js